src/main/java/core/basesyntax/services/DataFileReader.java

```diff
@@ -0,0 +1,27 @@
+package core.basesyntax.services;
+
+import com.opencsv.CSVReader;
+import java.io.FileReader;
+import java.io.IOException;
+import java.util.ArrayList;
+import java.util.List;
+
+public class DataFileReader {
+    private static final int PARTS_OF_LINE = 4;
+
+    public List<String[]> readDataFromFile(String inputFilePath) {
+        List<String[]> data = new ArrayList<>();
```

Comment:
I suggest we create separate model record/transaction with fields operation, quantity, name etc. So we will parse our data not to List<List<String>> but each row to this model

Reply:
I see your point, but exactly in my case I think it's a bit redundant. If I did this task other way, not with Maps, it would be fine, but in this realisation if I create separate model I need to parse String data into that model and then take Strings from that model and put them into my Map. I would like to avoid this extra convertation.
I agree with Alexandr about making Store, Model and Fruit classes is more flexible and gives more variants for future growth but exact for this task I think that my solution fits well.

Reply:
Point of this task is not to write code "that works ok now"  but to make code that would be easy to maintain in future, cause code maintenance is 60-80% of total cost and development is at most 20%. If you stick with String[] line approach whenever input format is changed, even if two fields have switched places you would need to make changes everywhere you line is used, cause index sequence will be changed, i your parse each line to specific model you will need to adjust only parsing, plus with external library for csv parsing you will be able to pars row into your model directly. And it improves readability as well.